src/sentry/integrations/github/client.py

```diff
@@ -55,8 +57,43 @@ def __repr__(self) -> str:
         return f"GithubRateLimitInfo(limit={self.limit},rem={self.remaining},reset={self.reset})"
 
 
+class GithubSetupApiClient(IntegrationProxyClient):
+    """
+    API Client that doesn't require an installation.
+    This client is used during integration setup to fetch data
+    needed to build installation metadata
+    """
+
+    base_url = "https://api.github.com"
+    integration_name = "github_setup"
+
+    def __init__(self, access_token: str | None = None, verify_ssl: bool = True):
+        super().__init__(verify_ssl=verify_ssl)
+        self.jwt = get_jwt()
+        self.access_token = access_token
+
+    @control_silo_function
+    def authorize_request(self, prepared_request: PreparedRequest) -> PreparedRequest:
+        if any(url in prepared_request.path_url for url in JWT_AUTH_ROUTES):
+            prepared_request.headers["Authorization"] = f"Bearer {self.jwt}"
+        else:
+            prepared_request.headers["Authorization"] = f"token {self.access_token}"
+
+        prepared_request.headers["Accept"] = "application/vnd.github+json"
+        return prepared_request
+
+    def get_installation_info(self, installation_id: int | str) -> dict[str, Any]:
+        """
+        https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-an-installation-for-the-authenticated-app
+        """
+        return self.get(f"/app/installations/{installation_id}")
+
+    def get_user_info(self):
+        return self.get("/user")
```

Comment:
this pattern was yoinked from https://github.com/getsentry/sentry/blob/32f4fbd216e1a1cebe26ed6db3420233250ba13b/src/sentry/integrations/gitlab/client.py#L34-L67